internal/cmd/satellites.go

```diff
@@ -0,0 +1,217 @@
+package cmd
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"cdr.dev/coder-cli/internal/x/xcobra"
+
+	"github.com/spf13/cobra"
+	"golang.org/x/xerrors"
+
+	"cdr.dev/coder-cli/coder-sdk"
+	"cdr.dev/coder-cli/pkg/clog"
+	"cdr.dev/coder-cli/pkg/tablewriter"
+)
+
+const (
+	satelliteKeyPath = "/api/private/satellites/key"
+)
+
+type satelliteKeyResponse struct {
+	Key         string `json:"key"`
+	Fingerprint string `json:"fingerprint"`
+}
+
+func satellitesCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "satellites",
+		Short: "Interact with Coder satellite deployments",
+		Long:  "Perform operations on the Coder satellites for the platform.",
+	}
+
+	cmd.AddCommand(
+		createSatelliteCmd(),
+		listSatellitesCmd(),
+		deleteSatelliteCmd(),
+	)
+	return cmd
+}
+
+func createSatelliteCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "create [name] [satellite_access_url]",
+		Args:  xcobra.ExactArgs(2),
+		Short: "create a new satellite.",
+		Long:  "Create a new Coder satellite.",
+		Example: `# create a new satellite
+
+coder satellites create eu-west https://eu-west-coder.com`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			var (
+				ctx       = cmd.Context()
+				name      = args[0]
+				accessURL = args[1]
+			)
+
+			client, err := newClient(ctx, true)
+			if err != nil {
+				return xerrors.Errorf("making coder client: %w", err)
+			}
+
+			sURL, err := url.Parse(accessURL)
+			if err != nil {
+				return xerrors.Errorf("parsing satellite access url: %w", err)
+			}
+			sURL.Path = satelliteKeyPath
+
+			// Create the http request.
+			req, err := http.NewRequestWithContext(ctx, http.MethodGet, sURL.String(), nil)
+			if err != nil {
+				return xerrors.Errorf("create satellite request: %w", err)
+			}
+			res, err := http.DefaultClient.Do(req)
+			if err != nil {
+				return xerrors.Errorf("doing satellite request: %w", err)
+			}
+			defer func() { _ = res.Body.Close() }()
+
+			if res.StatusCode > 299 {
+				return fmt.Errorf("unexpected status code %d: %+v", res.StatusCode, res)
+			}
+
+			var keyRes satelliteKeyResponse
+			if err := json.NewDecoder(res.Body).Decode(&keyRes); err != nil {
+				return xerrors.Errorf("decode response body: %w", err)
+			}
+
+			if keyRes.Key == "" {
+				return xerrors.New("key field empty in response")
+			}
+			if keyRes.Fingerprint == "" {
+				return xerrors.New("fingerprint field empty in response")
+			}
+
+			fmt.Printf(`The following satellite will be created:
+Name: %s
+
+Public Key: 
+%s
+
+Fingerprint:
+%s
+
+Do you wish to continue? (y)
```

Comment:
Typically having an option in parenthesis means that it's the default if you hit enter and don't type anything, I don't think your code does that.

Reply:
Should I do `(y/n)`? I still wanna prompt the user to hit `y`

Reply:
Yeah that's good. In that case you should also `response = strings.ToLower(strings.TrimSpace(response))` the response before checking